org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/diagnostics/LSPDiagnosticsToMarkers.java

```diff
@@ -166,12 +170,16 @@ public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
 
 				try {
 					for (Diagnostic diagnostic : newDiagnostics) {
-						Map<String, Object> markerAttributes = computeMarkerAttributes(document, diagnostic, resource);
-						resource.createMarker(markerType, markerAttributes);
+						if (resource.exists()) {
+							Map<String, Object> markerAttributes = computeMarkerAttributes(document, diagnostic, resource);
+							resource.createMarker(markerType, markerAttributes);
+						}
 					}
 					for (Entry<IMarker, Diagnostic> entry : toUpdate.entrySet()) {
-						Map<String, Object> markerAttributes = computeMarkerAttributes(document, entry.getValue(), resource);
-						updateMarker(markerAttributes, entry.getKey());
+						if (resource.exists()) {
```

Comment:
couldn't it be `if (entry.getKey().exists())` here instead?

Reply:
Yes, you're right.